client/my-sites/plans-features-main/index.tsx

```diff
@@ -548,7 +548,7 @@ const PlansFeaturesMain = ( {
 			};
 		}
 
-		if ( sitePlanSlug && isFreePlan( sitePlanSlug ) && intent !== 'plans-p2' ) {
+		if ( sitePlanSlug && isFreePlan( sitePlanSlug ) && intentFromProps !== 'plans-p2' ) {
```

Comment:
Hey @jeyip any reason you didn't include this prop in the deps array 🤔

Reply:
It looks like future Dehan did us a favor and already added `intentFromProps` to the deps array in https://github.com/Automattic/wp-calypso/pull/84515. Thanks future Dehan :D

https://github.com/Automattic/wp-calypso/blob/b4313a87eefb0b820d34caf60d093bf73dd783b1/client/my-sites/plans-features-main/index.tsx#L624-L636